### ISSUE TITLE
Openzaak management portal applicatie creation

### ISSUE THREAD

**Issue by aurimasroxit97** [open-zaak/open-zaak#835]
Hi. I am looking for a confirmation whether this is a bug or not.

**Describe the bug**
When creating an application (API autorisaties -> Applicaties), it is possible to create an application with _heeftAlleAutorisaties_ set to false + without any authorizations defined. 
This seems to conflict with the following business rule in the standard:

_Indien heeftAlleAutorisaties gezet is, dan moet je autorisaties leeg (of weg) laten.
Indien je autorisaties meegeeft, dan moet heeftAlleAutorisaties de waarde false hebben of weggelaten worden_

**To Reproduce**
1. Go to API autorisaties -> Applicaties module, click on "APPLICATIE TOEVOEGEN"
2. Create an application with heeftAlleAutorisaties: false and do not define any authorizations.

**Expected behavior**
Creating an application without any authorizations should not be allowed, as it is said in the standard?

**Comment by sergei-maertens** [comment]
This is a tricky one - we cannot enforce this validation since the authorizations can only be created once the application record has been created. And that requires creating the application with `heeftAlleAutorisaties` unchecked (and no authorizations).

Maybe adding a flag in the admin list to show invalid applications is something that we can do to draw attention.

So, unmarking this as a bug, but I'll add the enhancement label.

**Comment by aurimasroxit97** [comment]
@sergei-maertens Thank you for your response. Can you explain how do you submit such an applicatie like you described to openzaak autorisaties API?
I tried doing a HTTP POST on autorisaties/api/v1/applicaties
with this body:        
{
            "clientIds": ["myclientid"],
            "label": "test2",
            "heeftAlleAutorisaties": false,
            "autorisaties": []
 }

And it returns 400 missing-authorizations error. However, I can clearly see that records like these exist in the API. So, how do you submit an applicatie without autorisaties and "heeftAlleAutorisaties": false ?

**Comment by sergei-maertens** [comment]
@aurimasroxit97 

```json
{
    "clientIds": ["myclientid"],
    "label": "test2",
    "heeftAlleAutorisaties": false,
    "autorisaties": []
}
```
is indeed not valid _in the API_. Here we can validate that entiry entity - the API call receives all the information.

Note that the admin interface is not built on top of the API - it's a built-in interface provided by the framework on top of the database. This is the source of the different behaviour.

An unintented side-effect is that those incomplete applications show up in the API if created through the admin. I think we should exclude those invalid applications from showing up in the API, until they are valid.

To be absolutely clear:

> So, how do you submit an applicatie without autorisaties and "heeftAlleAutorisaties": false ?

Using the API - you don't.

Re-adding the `bug` label to account for the confusing behaviour in the API because of admin actions, I think we should tackle this in 1.4.0.

**Comment by aurimasroxit97** [comment]
@sergei-maertens, To us it seems now that Open zaak does not comply to the ZGW standards. Is this true? Please also see our question https://github.com/VNG-Realisatie/gemma-zaken/issues/1761

**Comment by sergei-maertens** [comment]
We pass the [test-suites of VNG API test platform](https://github.com/open-zaak/open-zaak/runs/1838832502?check_suite_focus=true), so for everything described in there, we do comply with the standard.

Of course, this test suite is also not perfect and their is pointed out behaviour that can lead to inconsistent data which does not comply: you can create an application without superuser and without authorizations in the Open Zaak admin, which currently causes it to show up in the Autorisaties API and yields unexpected results. This will be fixed in the next relese (patch or feature).

With regard to the linked upstream issue - OAS has the concept of `operationId` to essentially label the combination of URL path + HTTP method. To the best of my knowledge, Open Zaak complies with the standard. If it doesn't in places, please point them out - as those are considered bugs that need to be fixed.

I'll reply in the upstream issue with anything I can clarify :)

**Comment by sergei-maertens** [comment]
Fix and better feedback PR: #867
